appOPHD/States/ColonyShip.cpp

```diff
@@ -0,0 +1,24 @@
+#include "ColonyShip.h"
+
+#include "appOPHD/Constants/Strings.h"
+
+ColonyShip::ColonyShip()
+{}
+
+
+ColonyShip::ColonyShip(const ColonyShipData& colonyShipData) :
+	mColonyShipData(colonyShipData)
+{}
+
+
+void ColonyShip::onTurn()
+{
+	if (mColonyShipData.turnsOfManeuveringFuel > 0) { --mColonyShipData.turnsOfManeuveringFuel; }
+
+	if (mColonyShipData.turnsOfManeuveringFuel == 0)
+	{
+		mCrashData = mColonyShipData;
+		mColonyShipData.cargoLanders = 0;
+		mColonyShipData.colonistLanders = 0;
+	}
+}
```

Comment:
Just noting this section seems to handle this event differently than we normally do in other parts of the code. Other sections would tend to use `Signal` or `Delegate` to report an event. Here the `std::optional` field `mCrashData` is being set, which is later polled.

A potential alternative might be to return crash data from this function.

No change needed here. Just pointing out the style is different. We should probably try this out so we have some experience with this style.